cloud/clouds.go

```diff
@@ -281,6 +282,11 @@ func ParseCloudMetadata(data []byte) (map[string]Cloud, error) {
 		return nil, errors.Annotate(err, "cannot unmarshal yaml cloud metadata")
 	}
 
+	if err := validateCloudSet(data); err != nil {
+		logger := loggo.GetLogger("juju.cloud")
+		logger.Warningf("The imported cloud definition contains errors:\n%s", err.Error())
```

Comment:
I think this is the wrong place to be translating the error to a warning; IMO that belongs in the CLI code, if anywhere. This function may be called by other code, and there will be no way for that code to know that the returned cloud definitions are invalid.

I think there are two viable options:
 1. parse the valid cloud definitions and return those, a map of cloud names to errors, and a special error informing the caller that there were some errors
 2. error out immediately if any of the cloud definitions are invalid

The former *might* be helpful for the CLI (some invalid clouds don't prevent all from being parsed), but I'm not that convinced it would be.